src/screens/splash.rs

```diff
@@ -135,8 +176,34 @@ fn tick_splash_timer(time: Res<Time>, mut timer: ResMut<SplashTimer>) {
     timer.0.tick(time.delta());
 }
 
-fn check_splash_timer(timer: ResMut<SplashTimer>, mut next_screen: ResMut<NextState<Screen>>) {
-    if timer.0.just_finished() {
+fn check_splash_timer(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut timer: ResMut<SplashTimer>,
+    mut next_screen: ResMut<NextState<Screen>>,
+    mut splash_screens: ResMut<SplashScreenImageList>,
+    containers: Query<Entity, With<SplashScreenContainer>>,
+) {
+    if !timer.0.just_finished() {
+        return;
+    }
+
+    // try to get  the next splash screen. If there isn't one, we move on to the next screen
+    if let Some(next_splash_image) = splash_screens.0.pop_front() {
+        // despawn other splash images
+        if let Ok(container) = containers.get_single() {
```

Comment:
`let-else` here as well.

Reply:
Thanks - I forgot there is `let-else` 😁 

The container *should* exist, so what do you think about just doing `get()` here?

Reply:
I'm an `unwrap` hater so I'd say just do `let-else`, and in my own code I would do `let container = r!(containers.get_single());` using the shorthand `or_return!` macro from `tiny_bail`.

Reply:
Oh actually after a second look, I would rename `containers` -> `container_query` and make this a `for` loop. No need to assume / check / whatever that there's exactly one matching entity and special-case this, better to handle multiple entities for free.